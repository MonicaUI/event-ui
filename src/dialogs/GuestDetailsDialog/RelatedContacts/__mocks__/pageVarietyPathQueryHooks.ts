import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        guestRegistration {
          addGuestFromRelatedContacts {
            style {
              relatedContactsModalStyles
            }
            relatedContactsModalText
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useGuestRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
