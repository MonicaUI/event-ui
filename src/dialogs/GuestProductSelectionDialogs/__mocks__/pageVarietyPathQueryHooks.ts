import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        registration {
          admissionItems {
            display {
              fees
            }
          }
          sessions {
            display {
              fees
            }
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
