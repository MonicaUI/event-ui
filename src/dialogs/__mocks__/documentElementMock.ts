export const getMockedMessageContainer = (): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    <div name="messageContainer" tabIndex=-1>message container</div>
    </div>
  `;
  const mockMessageContainerFocus = jest.fn();
  const mockMessageContainer = document.getElementsByName('messageContainer')[0];
  mockMessageContainer.focus = mockMessageContainerFocus;
  return mockMessageContainer;
};

export const getMockedFieldInputs = (fieldNames: $TSFixMe): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    ${fieldNames.map(fieldName => `<input name=${fieldName}>input-${fieldName}</input>`)}
    </div>
  `;
  const mockInputFocus = jest.fn();
  const mockInputContainers = [];
  fieldNames.forEach(fieldName => {
    const mockInputContainer = document.getElementsByName(fieldName)[0];
    mockInputContainer.focus = mockInputFocus;
    mockInputContainers.push(mockInputContainer);
  });
  return mockInputContainers;
};
