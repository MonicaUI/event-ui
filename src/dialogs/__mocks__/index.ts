/* eslint-env jest */
import noopAction from '../../testUtils/noopAction';

export const openAlreadyRegisteredDialog = jest.fn(noopAction);

export const openDeclineRegistrationDialog = jest.fn(noopAction);

export const openStartNewRegistrationDialogFromPageLanding = jest.fn(noopAction);

export const openStartNewRegistrationDialogDuringRegistration = jest.fn(noopAction);

export const startNewRegistrationAndNavigateToRegistration = jest.fn(noopAction);

export const openEventWaitlistDialog = jest.fn(noopAction);

export const openEventTemporaryClosedErrorDialog = jest.fn(noopAction);

export const openNoAdmissionItemAvailableForRegistrationTypeDialog = jest.fn(noopAction);

export const openEventStatusDialog = jest.fn(noopAction);

export const openPrivateEventErrorDialog = jest.fn(noopAction);

export const openRegistrationInOtherTabDialog = jest.fn(noopAction);

export const openKnownErrorDialog = jest.fn(noopAction);

export const openPaymentProcessingErrorDialog = jest.fn(noopAction);

export const openInformationAlreadyUsedDialog = jest.fn(noopAction);

export const openPartialRegistrationConfirmationDialog = jest.fn(noopAction);

export const openPreviewModeWarningDialog = jest.fn(noopAction);

export const openCapacityReachedDialog = jest.fn(noopAction);

export const openGroupCancelRegistrationDialog = jest.fn(noopAction);

export const openSessionOverlapWarningDialog = jest.fn(noopAction);

export const openGroupMemberRemoveDialog = jest.fn(noopAction);

export const openSharePromptDialog = jest.fn(noopAction);

export const openConfirmationDialog = jest.fn(noopAction);

export const openGuestProductSelectionDialog = jest.fn(noopAction);

export const openGroupRegistrationTypeDialog = jest.fn(noopAction);

export const openGuestDetailsDialog = jest.fn(noopAction);

export const openGuestRemoveDialog = jest.fn(noopAction);

export const openContactPlannerDialog = jest.fn(noopAction);

export const openCancelRegistrationSuccessConfirmationDialog = jest.fn(noopAction);

export const openCancelRegistrationDialog = jest.fn(noopAction);

export const openShoulderDateApprovalWarningDialog = jest.fn(noopAction);

export const openTransactionInProcessingErrorDialog = jest.fn(noopAction);
