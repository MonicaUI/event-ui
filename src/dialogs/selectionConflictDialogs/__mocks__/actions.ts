export const returnToProcessStart = jest.fn(() => () => {});
