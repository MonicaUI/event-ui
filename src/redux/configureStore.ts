import analyticsConfig from './analytics';
import reducer from './reducer';

// using require instead of import for conditional compilation;
let createStoreWithMiddleware;
if (process.env.NODE_ENV === 'production') {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  createStoreWithMiddleware = require('nucleus-guestside-site/src/redux/prodCreateStoreWithMiddleware').default;
} else {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  createStoreWithMiddleware = require('nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware').default;
}

export default function configureStore(
  initialState: $TSFixMe,
  baseAnalyticsConfig = {},
  thunkExtraArgument?: $TSFixMe
): $TSFixMe {
  return createStoreWithMiddleware(reducer, initialState, {
    analyticsConfig: analyticsConfig(initialState, baseAnalyticsConfig),
    thunkExtraArgument
  });
}
