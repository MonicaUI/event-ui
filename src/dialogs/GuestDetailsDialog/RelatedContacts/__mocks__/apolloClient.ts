import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type RelatedContactsDialogMockData = {
  relatedContactsModalStyles: unknown;
  relatedContactsModalText: unknown;
};

export const getApolloClientMocks = (
  state: RelatedContactsDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { relatedContactsModalStyles = {}, relatedContactsModalText = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              guestRegistration: {
                addGuestFromRelatedContacts: {
                  style: {
                    relatedContactsModalStyles
                  },
                  relatedContactsModalText
                }
              }
            }
          }
        }
      }
    }
  ];
};
