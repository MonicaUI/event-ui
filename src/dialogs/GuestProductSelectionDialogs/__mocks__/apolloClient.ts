import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type GuestProductSelectionDialogMockData = {
  displayAdmissionItemsFees: unknown;
  displaySessionsFees: unknown;
};

export const getApolloClientMocks = (
  state: GuestProductSelectionDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { displayAdmissionItemsFees = {}, displaySessionsFees = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              registration: {
                admissionItems: {
                  display: {
                    fees: displayAdmissionItemsFees
                  }
                },
                sessions: {
                  display: {
                    fees: displaySessionsFees
                  }
                }
              }
            }
          }
        }
      }
    }
  ];
};
