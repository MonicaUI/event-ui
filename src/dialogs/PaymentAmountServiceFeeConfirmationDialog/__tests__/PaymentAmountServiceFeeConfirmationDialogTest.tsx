import React from 'react';
import DialogContainer from 'nucleus-guestside-site/src/containers/DialogContainer';
import configureStore from '../../../redux/configureStore';
import { act } from 'react-dom/test-utils';
import { MockedProvider } from '@apollo/client/testing';
import { merge } from 'lodash';
import { openPaymentAmountServiceFeeConfirmationDialog } from '..';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import EventSnapshot from '../../../../fixtures/EventSnapshot.json';
// eslint-disable-next-line jest/no-mocks-import
import { RegCartClient } from '../../../widgets/PaymentWidget/__mocks__/regCartClient';
// eslint-disable-next-line jest/no-mocks-import
import { MOCK_GET_REG_CART_PRICING } from '../../../widgets/PaymentWidget/__mocks__/useCachedRegCartPricing';
// eslint-disable-next-line jest/no-mocks-import
import { getApolloClientMocks } from '../../../widgets/PaymentWidget/__mocks__/apolloClient';
jest.mock('../../../widgets/PaymentWidget/getRegCartPricingAction', () => ({
  __esModule: true,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  default: async (state, _) => {
    return state;
  }
}));
jest.mock('../../../widgets/PaymentWidget/useCachedRegCartPricing', () => ({
  ...jest.requireActual<$TSFixMe>('../../../widgets/PaymentWidget/useCachedRegCartPricing'),
  __esModule: true,
  getTravelCartForQuery: () => {
    return JSON.stringify({});
  },
  lastSavedRegCartForQuery: () => {
    return JSON.stringify({});
  }
}));
jest.mock('../../../widgets/PaymentWidget/regCartPricing', () => ({
  ...jest.requireActual<$TSFixMe>('../../../widgets/PaymentWidget/regCartPricing'),
  getRegCartPaymentInfo: () => {
    return {};
  },
  getRegCartPricingGQL: () => {
    return MOCK_GET_REG_CART_PRICING;
  }
}));

const regCartPricing = {
  netFeeAmountRefund: '0',
  inviteeTypeServiceFeePricingRefunds: {},
  eventRegistrationPricings: [
    {
      netFeeAmountRefund: '0',
      eventRegistrationId: '00000000-0000-0000-0000-000000000001',
      netFeeAmountChargeWithPaymentTypeServiceFee: '145.2',
      productPricings: [],
      productFeeAmountCharge: '100',
      chargeRevenueShareFees: {
        salesTax: null,
        fees: null
      },
      merchantProcessingFees: {
        paymentAmount: '0',
        fee: '0'
      },
      productSubTotalAmountCharge: '100',
      netFeeAmountCharge: '121',
      transactionIdToRefundAmounts: {},
      refundMerchantProcessingFee: null,
      productSubTotalAmountRefund: '0',
      refundRevenueShareFees: {
        salesTax: null,
        fees: null
      },
      productFeeAmountRefund: '0'
    }
  ],
  productFeeAmountCharge: '100',
  netFeeAmountChargeWithPaymentAmountServiceFee: 145.2,
  paymentTypeServiceFeePricingRefunds: {},
  paymentTypeServiceFeePricingCharges: {
    '0e3d7926-a956-4158-8908-f76a0aa0a3e5': {
      primaryRegToOrderDetailIds: {
        '00000000-0000-0000-0000-000000000001': '09c2deca-f85a-4428-8006-952817d19131'
      },
      appliedOrder: '1',
      id: '0e3d7926-a956-4158-8908-f76a0aa0a3e5',
      feeApplyType: '1',
      totalPaymentTypeServiceFeeAmount: '12.1',
      inviteeBreakdowns: [
        {
          amount: '12.1',
          attendeeType: 'ATTENDEE',
          attendeeId: null,
          eventRegistrationId: '00000000-0000-0000-0000-000000000001',
          orderDetailTaxFeeId: null
        }
      ]
    },
    'c05fd891-1f48-4f03-a416-3f3db7902af4': {
      primaryRegToOrderDetailIds: {
        '00000000-0000-0000-0000-000000000001': 'c4198c04-495e-4500-a0cf-2ba6ed29655a'
      },
      appliedOrder: '4',
      id: 'c05fd891-1f48-4f03-a416-3f3db7902af4',
      feeApplyType: '1',
      totalPaymentTypeServiceFeeAmount: '12.1',
      inviteeBreakdowns: [
        {
          amount: '12.1',
          attendeeType: 'ATTENDEE',
          attendeeId: null,
          eventRegistrationId: '00000000-0000-0000-0000-000000000001',
          orderDetailTaxFeeId: null
        }
      ]
    }
  },
  productSubTotalAmountCharge: '100',
  inviteeTypeServiceFeePricingCharges: {},
  netFeeAmountCharge: '121',
  taxPricingCharges: {},
  transactionIdToRegCartPricingRefund: {},
  taxPricingRefunds: {},
  regCartId: 'ed86b8ee-8982-4395-ae24-336a3dc8e234',
  productSubTotalAmountRefund: '0',
  productFeeAmountRefund: '0'
};

const initialState = {
  text: {
    translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx),
    resolver: {
      date: () => 'some date',
      currency: x => x
    }
  },
  website: EventSnapshot.eventSnapshot.siteEditor.website,
  regCartPricing: {
    ...regCartPricing
  },
  registrationForm: {
    regCart: {
      regCartId: '00001',
      eventRegistrations: {
        'b18958b2-bf64-4e2e-9c07-e13fd4b0f314': {
          attendeeType: 'ATTENDEE',
          eventRegistrationId: 'b18958b2-bf64-4e2e-9c07-e13fd4b0f314'
        }
      }
    },
    regCartPayment: {}
  },
  clients: {
    regCartClient: new RegCartClient()
  },
  regCartStatus: {
    lastSavedRegCart: {}
  },
  travelCart: {
    cart: {}
  }
};

const waitWithAct = async () => {
  // Act is used because the Redux store is updated during wait
  return act(async () => {
    await new Promise(resolve => setTimeout(resolve, 0));
  });
};

let store;
const mountComponent = async (props = {}, optionalState = {}) => {
  const state = merge({}, initialState, optionalState);
  store = configureStore(state, {});

  const apolloClientMocks = getApolloClientMocks(state);

  const component = mount(
    <Provider store={store}>
      {/* @ts-expect-error ts-migrate(2322) FIXME: Type '{ children: Element; mocks: any; addTypeName... Remove this comment to see the full error message */}
      <MockedProvider mocks={apolloClientMocks} addTypeName={false}>
        <DialogContainer {...props} spinnerMessage="spinnerMessage" message="message" />
      </MockedProvider>
    </Provider>
  );
  // Wait for Apollo Client MockedProvider to render mock query results
  await waitWithAct();
  await component.update();
  return component;
};

describe('PaymentAmountServiceFeeConfirmationDialog', () => {
  test('matches snapshot when opened', async () => {
    const component = await mountComponent({}, {});
    store.dispatch(openPaymentAmountServiceFeeConfirmationDialog());
    await waitWithAct();
    await component.update();
    expect(component).toMatchSnapshot();
  });

  test('renders correct order total when payment credits adjusted', async () => {
    // a total of 100 credits applied
    const customState = {
      regCartPricing: {
        paymentCreditsForEventReg: {
          eventRegistration1: {
            creditsCharge: 25
          },
          eventRegistration2: {
            creditsCharge: 75
          }
        }
      }
    };

    const dialog = await mountComponent({}, customState);
    store.dispatch(openPaymentAmountServiceFeeConfirmationDialog());
    await waitWithAct();
    await dialog.update();

    const totalWithCreditsAdjusted = regCartPricing.netFeeAmountChargeWithPaymentAmountServiceFee - 100;
    const expectedTotalDueResx = `EventWidgets_PaymentWidget_TotalDue__resx:{"totalDue":${totalWithCreditsAdjusted}}`;
    const totalDueElement = dialog.find('[data-cvent-id="total-due-value"]');
    expect(totalDueElement).toHaveLength(1);
    expect(totalDueElement.props().children === expectedTotalDueResx).toBeTruthy();
  });

  test('click on cancel payment amount service fee confirmation dialog', async () => {
    const dialog = await mountComponent();
    store.dispatch(openPaymentAmountServiceFeeConfirmationDialog());
    await waitWithAct();
    await dialog.update();
    dialog.find('[data-cvent-id="no-submit-button"]').hostNodes().simulate('click');
    expect(dialog).toMatchSnapshot();
  });
});
