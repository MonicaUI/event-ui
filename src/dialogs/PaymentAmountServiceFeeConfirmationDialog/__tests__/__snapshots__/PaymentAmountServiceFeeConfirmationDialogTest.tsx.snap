// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentAmountServiceFeeConfirmationDialog click on cancel payment amount service fee confirmation dialog 1`] = `
<div
  class=""
>
  <div
    class=""
  />
  <span
    class="mockNucleusCoreTransition"
  />
</div>
`;

exports[`PaymentAmountServiceFeeConfirmationDialog matches snapshot when opened 1`] = `
<div
  class=""
>
  <div
    class=""
  />
  <span
    class="mockNucleusCoreTransition"
  >
    <div
      class=""
    >
      <div
        aria-label="Dialog"
        role="dialog"
        style="outline: none;"
        tabindex="0"
      >
        <div
          class=""
          style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
        >
          <div
            class=""
          />
          <div
            class=""
          >
            <div
              class="dialogContainer"
            >
              <div
                class="dialogChildren"
              >
                <div
                  data-cvent-id="payment-amount-service-fee-confirmation-dialog"
                >
                  <div>
                    <div
                      aria-describedby="dialogHeader"
                      class=""
                      style="background-color: rgb(221, 226, 230); border-radius: 0;"
                    >
                      <div
                        class=""
                        id="dialogHeader"
                        style="font-family: PT Sans, sans-serif; font-weight: 400; font-size: 1.875rem; color: rgb(4, 21, 50); line-height: 1.3; font-style: normal; background-color: transparent; border-radius: 0; padding: 0px 0px 0px 0px;"
                      >
                        <h2
                          class=""
                        >
                          <div
                            class=""
                            data-cvent-id="close"
                            role="button"
                            tabindex="0"
                          >
                            <span
                              class="container"
                            >
                              <span
                                aria-label="EventGuestSide_StandardDialog_FallbackText_Close__resx"
                                class=""
                                role="img"
                              />
                            </span>
                          </div>
                          ServiceFees_PaymentHeader__resx
                        </h2>
                      </div>
                    </div>
                    <div
                      class=""
                      name="messageContainer"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          data-cvent-id="confirmation-form"
                        >
                          <div
                            class="cancelRegistrationModalText"
                          >
                            <div
                              class="detailsText"
                            >
                              <div
                                class=""
                              >
                                <div
                                  class="leftLineItemStyle"
                                >
                                  <span
                                    class="instructionalText"
                                  >
                                    ServiceFees_PaymentInstruction__resx
                                  </span>
                                </div>
                                <br />
                                <br />
                                <hr
                                  class=""
                                />
                                <div
                                  class="rightLineItemStyle"
                                >
                                  <span
                                    class="totalAmountDue"
                                    data-cvent-id="total-due-value"
                                  >
                                    EventWidgets_PaymentWidget_TotalDue__resx:{"totalDue":145.2}
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                          <button
                            aria-label="GuestProductSelection_DialogCancelButtonText__resx"
                            class=""
                            data-cvent-id="no-submit-button"
                            title="GuestProductSelection_DialogCancelButtonText__resx"
                            type="button"
                          >
                            <span>
                              GuestProductSelection_DialogCancelButtonText__resx
                            </span>
                          </button>
                          <button
                            aria-label="GuestProductSelection_DialogConfirmButtonText__resx"
                            class=""
                            data-cvent-id="yes-submit-button"
                            title="GuestProductSelection_DialogConfirmButtonText__resx"
                            type="button"
                          >
                            <span>
                              GuestProductSelection_DialogConfirmButtonText__resx
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </span>
</div>
`;
